### PR TITLE
Update doc about implementation on sandbox

### PR DESCRIPTION
**Pull request by venkata-lakshmi-penna**
Signed-off-by: Venkata Lakshmi Dhatri Penna <venkata.lakshmi.dhatrix.penna@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
 `docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`
+
+## Deployment On Sandbox
+
+
+The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**
+
+#### Get Access To Sandbox
+
+To deploy the pipeline on test environment, You need to request for access to sanbox through **[LF Service Desk](https://jira.linuxfoundation.org/servicedesk/customer/portals)**
```
Comment (jamesrgregg) [inline README.md]: The official process to request access is found [on the lf-releng docs site](https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html)

Comment (jamesrgregg) [reply README.md]: https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox

Comment (jamesrgregg) [reply README.md]: Once the LF ID exists, the request can be made to add via JIRA ticket request.

File: README.md [hunk 2]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.
+
+#### Implementation
+
+Once you create a new jenkins file follow the procedure as above. In your github repository, your jenkinsfile name should be Jenkinsfile.sanbox. Since, the jenkins sanbox will recognizes only the file Jenkinsfile.sanbox to execute.
```
Comment (jamesrgregg) [inline README.md]: spelling eror - please change to correct spelling - "sandbox"

Comment (jamesrgregg) [reply README.md]: I think we wanted to get away from using a separate Jenkinsfile based on some initial testing.

File: README.md [hunk 3]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+For the EdgeX Foundry Project Org all the configurations such as Credentials to access the github repos were already provided. Once you add Jenkinsfile.sandbox to the repository, Now you scan the organization on jenkins UI to execute the jobs.
+
+![edgexjenkinsorg.png](images/edgexjenkinsorg.PNG)
+
+It already has set the behaviour to discover the branches, pull request from origin and pull requests from forks.
+
+#### Things to remember
+
+* All the Jobs on jenkins Sanbox are automatically deleted every Saturday at 08:00 UTC
```
Comment (jamesrgregg) [inline README.md]: Jenkins - not jenkins

File: README.md [hunk 4]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.
+
+#### Implementation
+
+Once you create a new jenkins file follow the procedure as above. In your github repository, your jenkinsfile name should be Jenkinsfile.sanbox. Since, the jenkins sanbox will recognizes only the file Jenkinsfile.sanbox to execute.
+
+#### Configurations on Jenkins UI
+
+For the EdgeX Foundry Project Org all the configurations such as Credentials to access the github repos were already provided. Once you add Jenkinsfile.sandbox to the repository, Now you scan the organization on jenkins UI to execute the jobs.
```
Comment (jamesrgregg) [inline README.md]: Jenkins

File: README.md [hunk 5]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+![edgexjenkinsorg.png](images/edgexjenkinsorg.PNG)
+
+It already has set the behaviour to discover the branches, pull request from origin and pull requests from forks.
+
+#### Things to remember
+
+* All the Jobs on jenkins Sanbox are automatically deleted every Saturday at 08:00 UTC
+* Project configuration files and credentials are not loaded into the system
+* Jenkins nodes have the same OpenStack configuration as the production instance with    minor differences.
```
Comment (jamesrgregg) [inline README.md]: why mention the "minor differences"?

**Review by jamesrgregg — CHANGES_REQUESTED** [review]
I see a need for some changes please.  I also heard from @ernestojeda that we may want to consider a different approach so we don't introduce two builds - one on Sandbox and one the Production environment for the same PR.

File: README.md (hunk 1 above)
Comment (venkata-lakshmi-penna) [reply README.md]: I will update these changes mentioned.

File: README.md (hunk 5 above)
Comment (venkata-lakshmi-penna) [reply README.md]: fixed

File: README.md [hunk 6]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+#### Get Access To Sandbox
+
+The official process to request access sandbox is provided in the link **[Here](https://https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.
+
+#### Implementation
+
+Once you create a new Jenkins file follow the procedure as above. In your Github repository, your jenkinsfile name should be Jenkinsfile.sandbox. Since, the jenkins sanbox will recognizes only the file Jenkinsfile.sandbox to execute.
```
Comment (jamesrgregg) [inline README.md]: spelling - sandbox

Comment (jamesrgregg) [reply README.md]: capitalize Jenkins

**Review by jamesrgregg — CHANGES_REQUESTED** [review]
please make some additional changes..

File: README.md (hunk 6 above)
Comment (venkata-lakshmi-penna) [reply README.md]: i apologize i donno  missed this some how

Comment (venkata-lakshmi-penna) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (venkata-lakshmi-penna) [reply README.md]: what i understood is we using the same jenkinsfile for box prod and sandbox but we just changing the name of the file right!

File: README.md [hunk 7]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
 `docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`
+
+## Deployment On Sandbox
+
+
+The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**
+
+#### Get Access To Sandbox
+
+The official process to request access sandbox is provided in the link **[Here](https://https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**
```
Comment (vishwassp-hcl-intel) [inline README.md]: change as 'access request'

File: README.md [hunk 8]
```diff
@@ -271,3 +274,35 @@ ENTRYPOINT ["/go-binary"]
...
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.
+
+#### Implementation
+
+Once you create a new Jenkins file follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. Since, the Jenkins sandbox will recognizes only the file Jenkinsfile.sandbox to execute.
```
Comment (vishwassp-hcl-intel) [inline README.md]: single word 'Jenkinsfile'

File: README.md (hunk 7 above)
Comment (venkata-lakshmi-penna) [reply README.md]: fixed

File: README.md (hunk 8 above)
Comment (venkata-lakshmi-penna) [reply README.md]: fixed

File: README.md [hunk 9]
```diff
@@ -271,3 +274,33 @@ ENTRYPOINT ["/go-binary"]
...
 
 `docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`
+
+## Deployment On Sandbox
+
+The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**
+
+#### Get Access To Sandbox
+
+The official process to request for access to sandbox is provided in the link **[Here](https://https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**
```
Comment (eb-oss) [inline README.md]: The link URL is broken due to having "https://https://" at the beginning.

File: README.md [hunk 10]
```diff
@@ -271,3 +274,33 @@ ENTRYPOINT ["/go-binary"]
...
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.
```
Comment (eb-oss) [inline README.md]: I don't think this line should be here. The site linked above contains the needed info, and this is misleading. Yes, you'll use the same LFID to access the sandbox as prod, but this seems to imply that if you have access to prod, you immediately have sandbox access, which is not true. As the docs show, each user has to request access to the sandbox from LF.

File: README.md [hunk 11]
```diff
@@ -271,3 +274,33 @@ ENTRYPOINT ["/go-binary"]
...
+
+#### Implementation
+
+Once you create a new Jenkinsfile follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. Since, the Jenkins sandbox will recognizes only the file Jenkinsfile.sandbox to execute.
```
Comment (eb-oss) [inline README.md]:
```suggestion
Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will only execute the Jenkinsfile.sandbox file.```

File: README.md [hunk 12]
```diff
@@ -271,3 +274,33 @@ ENTRYPOINT ["/go-binary"]
...
+
+#### Configurations on Jenkins UI
+
+For the EdgeX Foundry Project Org all the configurations such as Credentials to access the github repos were already provided. Once you add Jenkinsfile.sandbox to the repository, Now you scan the organization on Jenkins UI to execute the jobs.
+
+![edgexjenkinsorg.png](images/edgexjenkinsorg.PNG)
+
+It already has set the behaviour to discover the branches, pull request from origin and pull requests from forks.
```
Comment (eb-oss) [inline README.md]:
```suggestion
The Jenkins pipeline job will automatically discover branches and pull requests for the repository.
```

**Review by eb-oss — CHANGES_REQUESTED** [review]
After updating my PR to edgex-go which has a Jenkinsfile.sandbox, I'm not seeing the job get created in the sandbox. Does the pipeline need to be recreated each week, due to the sandbox wiping? If so, we should automate that, so that it is always recreated after the sandbox is wiped. But if we don't automate it, the process of recreating the multibranch pipeline should be documented.

**Review by ernestojeda — APPROVED** [review]

**Comment by eb-oss** [comment]
@ernestojeda There are multiple issues with this change. Please don't merge as-is.

File: README.md (hunk 9 above)
Comment (venkata-lakshmi-penna) [reply README.md]: fixed

File: README.md (hunk 10 above)
Comment (venkata-lakshmi-penna) [reply README.md]: Updated.

File: README.md (hunk 11 above)
Comment (venkata-lakshmi-penna) [reply README.md]: Updated

File: README.md (hunk 12 above)
Comment (venkata-lakshmi-penna) [reply README.md]: Updated

File: README.md [hunk 13]
```diff
@@ -271,3 +274,41 @@ ENTRYPOINT ["/go-binary"]
...
+
+#### Get Access To Sandbox
+
+The official process to request for access to sandbox is provided in the link **[Here](https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**
+
+If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox, but before each other must request the access to sandbox by creating an LF ticket at **[LFServicedesk](https://jira.linuxfoundation.org/servicedesk/customer/portals)**
+
+#### Implementation
+
+Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will execute only the file Jenkinsfile.sandbox file.
```
Comment (eb-oss) [inline README.md]:
```suggestion
Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will execute only the Jenkinsfile.sandbox file.
```

**Review by eb-oss — CHANGES_REQUESTED** [review]
Just one more fix required; there's a redundant word in one of your changes.

File: README.md (hunk 13 above)
Comment (venkata-lakshmi-penna) [reply README.md]: fixed it. Thanks

**Review by eb-oss — DISMISSED** [review]

**Review by jamesrgregg — CHANGES_REQUESTED** [review]
image does not display correctly and looks like it wasn't committed and the file extension should be .png not .PNG once committed.

**Comment by eb-oss** [comment]
> image does not display correctly and looks like it wasn't committed and the file extension should be .png not .PNG once committed.

@jamesrgregg I agree with changing the extension from .PNG to .png, but otherwise, everything seems fine to me. After pulling the branch locally, I can see the image in the document just fine.

**Review by jamesrgregg — APPROVED** [review]

**Pull request merged**